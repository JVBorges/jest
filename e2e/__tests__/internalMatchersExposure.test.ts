/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import {extractSummary} from '../Utils';
import runJest from '../runJest';

test('error when internal matcher inside a custom matcher fails', () => {
  const {exitCode, stderr} = runJest('internal-matcher-exposure');
  const {rest} = extractSummary(stderr);

  expect(rest).toMatchSnapshot();
  expect(exitCode).toBe(1);
});
