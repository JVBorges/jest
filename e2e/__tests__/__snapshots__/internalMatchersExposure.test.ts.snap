// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`error when internal matcher inside a custom matcher fails 1`] = `
"FAIL __tests__/index.test.js
  ✓ should pass
  ✕ internal match should not pass

  ● internal match should not pass

    expected 1 to be equal to 2

      29 |
      30 | it('internal match should not pass', () => {
    > 31 |   expect(1).toPass(2);
         |             ^
      32 | });
      33 |

      at Object.toPass (__tests__/index.test.js:31:13)"
`;
